### ISSUE TITLE
Server App Blazor: UI not refreshed after adding/removing CSS classes to an element. 

### ISSUE THREAD

**Issue by ForerunnerG34** [dotnet/aspnetcore#28294]
### Describe the bug
Add or remove programmatically a CSS class to an HTML element using C#. The DOM is updated however the UI is not updated with the CSS styles; it keeps using the previous style until you refresh the page or change anything in the Dev tools console.

### To Reproduce

Code sample:
```
 @foreach(TransactionsGroup dateGroup in group.DateGroup )
                {
                    <tr class="sub-group">
                        <td colspan="2" class="ms-font-l font-w-600">@dateGroup.Name</td>
                    </tr>
                    @for(int i=0; i < dateGroup.Transactions.Count ; i++)
                    {                
                    Transaction transaction = dateGroup.Transactions[i];

                    <tr class="transactionItem @(i==0 ? "first" : string.Empty) @(i==dateGroup.Transactions.Count-1 ? "last" : string.Empty) @(transaction.Id == SelectedId ? "selected" : string.Empty)"
                        @onclick="(() => SelectTransaction(transaction.Id))">
                        <td class="name">
                            <div class="ms-font-m-plus font-w-600">@transaction.Name</div>
                            <div class="ms-fontColor-neutralTertiary ms-font-s-plus">12:00</div>
                        </td>
                        <td class="amount">
                            <div class="ms-font-m-plus font-w-600">@transaction.Amount</div>
                            <div class="ms-fontColor-neutralTertiary ms-font-s-plus">5000.00</div>
                        </td>
                    </tr>
                    }                
                }
```
Explanation: 

If a TR has the "first" class it will round the top corners, if it has that "last" class it will round the bottom corners, if it has no class it will have no round corners.

Steps:
Change the "group" model to a new set of transactions. the HTML is rendered again with the correct classes; however the UI shows the previous styles with a new set of data.

### Exceptions (if any)

Platform: .Net 5

**Comment by ForerunnerG34** [comment]
I discovered a workaround: in the CSS class add a property for background-color, set any color you want or set it to transparent, that somehow forces Blazor to refresh the UI correctly.

**Comment by javiercn** [comment]
@ForerunnerG34 thanks for contacting us.

There's nothing especial about how Blazor handles CSS, it's whatever the browser does. I'm not sure why you are observing the behavior you mention, but it can be due to the way diffing works. If you have a list of elements to render and they have an identifiable key, I suggest you use [@key](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-5.0#use-key-to-control-the-preservation-of-elements-and-components) to see if that solves your issue.

**Comment by CesarIntriago1980** [comment]
Hi @javiercn, is a very weird behavior, the markup was always updated correctly but not the UI, you had to resize the window just a little bit or open dev tools and then immediately the UI would refresh. It was very annoying, after multiple tests I discovered that adding any background-color would force the UI the correctly represent what is in the markup. My code uses animations in the CSS and sets CSS classes progrmarically, maybe that causes issues with the diff. I tried setting a random GUID as the ID of every row but had no effect, I will try @key just to see it it changes something.  So far the background-color did the trick.
